src/gt4py/next/ffront/foast_to_itir.py

```diff
@@ -318,12 +318,22 @@ def visit_Call(self, node: foast.Call, **kwargs) -> itir.Expr:
     def _visit_astype(self, node: foast.Call, **kwargs) -> itir.FunCall:
         assert len(node.args) == 2 and isinstance(node.args[1], foast.Name)
         obj, dtype = node.args[0], node.args[1].id
-
-        # TODO check that we test astype that results in a itir.map_ operation
-        return self._map(
-            im.lambda_("it")(im.call("cast_")("it", str(dtype))),
-            obj,
-        )
+        if isinstance(obj, foast.TupleExpr):
+            casted_elements = []
+            for _, element in enumerate(obj.elts):
```

Comment:
if you don't use both elements of enumerate, you can remove it and directly do:
```suggestion
            for element in obj.elts:
```